### PR TITLE
Retry logic and update engine name in django

### PR DESCRIPTION
**Pull request by apantel**

### CODE REVIEW

File: python/django/cockroach_example/views.py [hunk 1]
```diff
@@ -2,16 +2,34 @@
...
 class PingView(View):
     def get(self, request, *args, **kwargs):
         return HttpResponse("python/django", status=200)
 
+def retry_on_exception(num_retries=3, on_failure=HttpResponse(status=500), delay_=0.5, backoff_=1.5):
+    def retry(view):
+        def wrapper(*args, **kwargs):
+            delay = delay_
+            for i in range(num_retries):
+                try:
```
Comment (rohany) [inline python/django/cockroach_example/views.py]: for an error to get raised, you're going to need to wrap the execution of `view` in a `with transaction.atomic():` (`from django.db import transaction`).

Are we trying to support nested retries here? if so, this won't work right?

File: python/django/cockroach_example/views.py [hunk 2]
```diff
@@ -21,13 +51,17 @@ def get(self, request, id=None, *args, **kwargs):
             customers = list(Customers.objects.filter(id=id).values())
         return JsonResponse(customers, safe=False)
 
+    @retry_on_exception(3)
+    @atomic
```
Comment (rohany) [inline python/django/cockroach_example/views.py]: I think the execution of view needs an atomic, not the view itself. Or does this first wrap the function with retry execption, and then atomic, so its doing the same thing?

Comment (apantel) [reply python/django/cockroach_example/views.py]: It wraps the function with atomic, and then wraps the result in retry.

**Comment by rohany** [comment]
The code LGTM -- have you tested to see that this works when a transaction is actually retried?

**Comment by apantel** [comment]
I wasn't able to force a transaction error, but I faked a random error in Django and it worked.

**Comment by apantel** [comment]
I don't think this addresses #87. I have not successfully run the django test suite against cockroachdb.

**Comment by rafiss** [comment]
I don't think this repo has bors, so you can just click the merge button when ready.

**Pull request merged**